base/server/src/com/netscape/cmscore/apps/CMSEngine.java

```diff
@@ -186,33 +186,14 @@ public String getInstanceDir() {
         return instanceDir;
     }
 
-    public static IPasswordStore getPasswordStore(String id, PropConfigStore cs)
-            throws EBaseException {
-        String pwdClass = null;
-        String pwdPath = null;
-
-        if (NuxwdogUtil.startedByNuxwdog()) {
-            pwdClass = NuxwdogPasswordStore.class.getName();
-            // note: pwdPath is expected to be null in this case
-        } else {
-            pwdClass = cs.getString("passwordClass");
-            pwdPath = cs.getString("passwordFile", null);
-        }
-
-        try {
-            IPasswordStore ps = (IPasswordStore) Class.forName(pwdClass).newInstance();
-            ps.init(pwdPath);
-            ps.setId(id);
-            return ps;
-        } catch (Exception e) {
-            logger.error("Cannot get password store: " + e);
-            throw new EBaseException(e);
-        }
-    }
-
     public synchronized IPasswordStore getPasswordStore() throws EBaseException {
         if (mPasswordStore == null) {
-            mPasswordStore = getPasswordStore(instanceId, mConfig);
+            try {
+                mPasswordStore =
+                    IPasswordStore.getPasswordStore(instanceId, mConfig.getProperties());
```

Comment:
There might be some (not sure how significant) performance impact since `PropConfigStore.getProperties()` has to enumerate all properties and put them into a `Map`. If this method is used in frequently used code, the performance impact might be noticeable. If this is a concern, we probably can call `PropConfigStore.getSource()` instead and handle the return value as a `Hashtable`.

Reply:
It's not a problem because `CMSEngine` caches the returned password store.  But I added a comment to that effect.